src/report/getReportBody.ts

```diff
@@ -84,7 +85,7 @@ export const getReportBody = (
             context.payload.after ??
             context.payload.pull_request?.head.sha ??
             context.sha,
-        dir: dir ? `for \`${dir}\`` : '',
+        name: customTitle || (dir ? `for \`${dir}\`` : ''),
```

Comment:
With this code, the report header will look like this (assumed that `custom-title: Custom title`):

## Coverage report for Custom title

If this is the result you want to achieve, then the option should be named as `tag`, because it does not replace the entire title, only the tag in it. Also, don't forget to replace `dir` variable here:
https://github.com/ArtiomTr/jest-coverage-report-action/blob/19e9e06d96fefa23aef7ee979af01f60d89fb256/src/report/getReportBody.ts#L81
Because this is the place, where action identifies comment to update in on each re-run.

However, if you want the report title to look like this (assumed that `custom-title: Custom title`):

## Custom title

Then, you should replace this line:
https://github.com/ArtiomTr/jest-coverage-report-action/blob/19e9e06d96fefa23aef7ee979af01f60d89fb256/src/format/REPORT.md#L3
With the following:
```md
## {{ title }}
```
Then move the old title into the `strings.json` file and import it from `getReportBody.ts`. And then change the code like this:

```ts
    const reportBody = insertArgs(REPORT, {
        head: getReportTag(dir),
        //  changes here ↓
        title: insertArgs(title || customTitle, {
            dir: dir ? `for \`${dir}\`` : '',
        }),
        body: reportContent,
        sha:
            context.payload.after ??
            context.payload.pull_request?.head.sha ??
            context.sha,
        dir: dir ? `for \`${dir}\`` : '',
    });
```

I don’t know which solution will suit you better, I am satisfied with both options, so choose what you like best!